src/cursor/cur_hs.c

```diff
@@ -169,7 +169,10 @@ __curhs_next(WT_CURSOR *cursor)
      */
     WT_ERR(__curhs_next_visible(session, hs_cursor));
 
+    if (0) {
 err:
+        cursor->reset(cursor);
```

Comment:
Why no `WT_TRET` here like the ones below?

Reply:
Nice catch